test/e2e/network_policy.go

```diff
@@ -150,246 +126,108 @@ var _ = framework.KubeDescribe("NetworkPolicy", func() {
-		defer func() {
-			By("Cleaning up the server.")
-			if err := f.ClientSet.Core().Pods(ns.Name).Delete(serverPod.Name, nil); err != nil {
-				framework.Failf("unable to cleanup pod %v: %v", serverPod.Name, err)
-			}
-		}()
-		defer func() {
-			By("Cleaning up the server's service.")
-			if err := f.ClientSet.Core().Services(ns.Name).Delete(service.Name, nil); err != nil {
-				framework.Failf("unable to cleanup svc %v: %v", service.Name, err)
-			}
-		}()
+		defer cleanupServerPodAndService(f, serverPod, service)
 		framework.Logf("Waiting for Server to come up.")
 		err := framework.WaitForPodRunningInNamespace(f.ClientSet, serverPod)
 		Expect(err).NotTo(HaveOccurred())
 
-		By("Testing pods can connect to both ports when isolation is off.")
+		By("Testing pods can connect to both ports when no policy is present.")
 		testCanConnect(f, ns, "basecase-reachable-80", service, 80)
 		testCanConnect(f, ns, "basecase-reachable-81", service, 81)
 
-		setNamespaceIsolation(f, ns, "DefaultDeny")
-
-		By("Testing pods cannot by default when isolation is turned on.")
-		testCannotConnect(f, ns, "basecase-unreachable-80", service, 80)
-		testCannotConnect(f, ns, "basecase-unreachable-81", service, 81)
-
 		By("Creating a network policy for the Service which allows traffic only to one port.")
-		policy := v1beta1.NetworkPolicy{
+		policy := &networking.NetworkPolicy{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: "allow-ingress-on-port-81",
 			},
-			Spec: v1beta1.NetworkPolicySpec{
+			Spec: networking.NetworkPolicySpec{
 				// Apply to server
 				PodSelector: metav1.LabelSelector{
 					MatchLabels: map[string]string{
 						"pod-name": serverPod.Name,
 					},
 				},
 				// Allow traffic only to one port.
-				Ingress: []v1beta1.NetworkPolicyIngressRule{{
-					Ports: []v1beta1.NetworkPolicyPort{{
+				Ingress: []networking.NetworkPolicyIngressRule{{
+					Ports: []networking.NetworkPolicyPort{{
 						Port: &intstr.IntOrString{IntVal: 81},
 					}},
 				}},
 			},
 		}
-		result := v1beta1.NetworkPolicy{}
-		err = f.ClientSet.Extensions().RESTClient().Post().Namespace(ns.Name).
-			Resource("networkpolicies").Body(&policy).Do().Into(&result)
-
+		policy, err = f.InternalClientset.Networking().NetworkPolicies(ns.Name).Create(policy)
 		Expect(err).NotTo(HaveOccurred())
-		defer func() {
-			By("Cleaning up the policy.")
-			if err = f.ClientSet.Extensions().RESTClient().
-				Delete().
-				Namespace(ns.Name).
-				Resource("networkpolicies").
-				Name(policy.Name).
-				Do().Error(); err != nil {
-				framework.Failf("unable to cleanup policy %v: %v", policy.Name, err)
-			}
-		}()
+		defer cleanupNetworkPolicy(f, policy)
 
+		By("Testing pods can connect only to the port allowed by the policy.")
 		testCannotConnect(f, ns, "client-a", service, 80)
 		testCanConnect(f, ns, "client-b", service, 81)
 	})
 
-	It("shouldn't enforce policy when isolation is off [Feature:NetworkPolicy]", func() {
```

Comment:
I think that's covered well enough by other tests? Eg, the first test checks that it can connect to the server before adding a default-deny policy, and can't connect after.